File: labs/solidworks/swex/pmpage/controls/dependencies/TagsAndDependencies.Enable.vb
Public Class DataModelEnable

    <ControlTag(NameOf(Enable))>
    Public Property Enable As Boolean

    <SelectionBox(swSelectType_e.swSelFACES)>
    <DependentOn(GetType(EnableDepHandler), NameOf(Enable))>
    Public Property Selection As IEntity

End Class

Public Class EnableDepHandler
    Inherits DependencyHandler

    Protected Overrides Sub UpdateControlState(ByVal control As IPropertyManagerPageControlEx, ByVal parents As IPropertyManagerPageControlEx())
        control.Enabled = CBool(parents.First().GetValue())
    End Sub
End Class

File: labs/solidworks/swex/pmpage/controls/dependencies/TagsAndDependencies.Enable.cs
public class DataModelEnable
{
    [ControlTag(nameof(Enable))]
    public bool Enable { get; set; }

    [SelectionBox(swSelectType_e.swSelFACES)]
    [DependentOn(typeof(EnableDepHandler), nameof(Enable))]
    public IEntity Selection { get; set; }
}

public class EnableDepHandler : DependencyHandler
{
    protected override void UpdateControlState(IPropertyManagerPageControlEx control, IPropertyManagerPageControlEx[] parents)
    {
        control.Enabled = (bool)parents.First().GetValue();
    }
}